core/src/main/java/org/apache/calcite/sql/dialect/OracleSqlDialect.java

```diff
@@ -126,6 +139,19 @@ public OracleSqlDialect(Context context) {
     return false;
   }
 
+  @Override public void unparseBoolLiteral(SqlWriter writer,
+      SqlBoolLiteral literal, int leftPrec, int rightPrec) {
+    Boolean value = (Boolean) literal.getValue();
+    if (value == null || majorVersion >= 23) {
+      super.unparseBoolLiteral(writer, literal, leftPrec, rightPrec);
+      return;
+    }
+    // low version oracle not support bool literal
+    SqlCall call =
```

Comment:
Creating a call seems overkill, you can just emit the expected output directly.
The expression emitted would also require parentheses.

Reply:
I think you should write `(1 = 1)`. You don't know what other operators will surround this expression

Reply:
This is a very good reminder, and I noticed that someone also pointed this out in [CALCITE-6480](https://issues.apache.org/jira/browse/CALCITE-6480?jql=project%20%3D%20CALCITE%20AND%20status%20%3D%20Open%20ORDER%20BY%20created%20DESC). I will fix this.